tests/python/test_cast.py

```diff
@@ -26,6 +27,29 @@ def func():
     assert z[None] == 1000
 
 
+@pytest.mark.parametrize('dtype', [ti.f32, ti.f64])
+def test_cast_default_fp(dtype):
+    ti.init(default_fp=dtype)
+
+    @ti.kernel
+    def func(x: int, y: int) -> float:
+        return ti.cast(x, float) * float(y)
+
+    assert func(23, 4) == pytest.approx(23.0 * 4.0)
+
+
+@pytest.mark.parametrize('dtype', [ti.i32, ti.i64])
+def test_cast_default_ip(dtype):
+    ti.init(default_ip=dtype)
+
+    @ti.kernel
+    def func(x: float, y: float) -> int:
+        return ti.cast(x, int) * int(y)
```

Comment:
This also tests `float` and `int` in arguments and return values.

Reply:
It would be nice to use a function that returns a value that overflows `i32` when your `int` is actually `i64`.